File: boj/Implemenation/16503/16503.py
import sys
input = sys.stdin.readline

def getRes(sa, op, sb):
  a = int(sa)
  b = int(sb)
  if(op == '+'): return a + b
  if(op == '-'): return a - b
  if(op == '*'): return a * b
  return -1*(abs(a)//abs(b)) if a*b < 0 else a//b

expr = input().split()

res1 = getRes(expr[0], expr[1], expr[2]);
res1 = getRes(str(res1), expr[3], expr[4]);

res2 = getRes(expr[2], expr[3], expr[4]);
res2 = getRes(expr[0], expr[1], str(res2));

print(min(res1, res2))
print(max(res1, res2))

File: boj/Implemenation/16503/16503.cpp
#include <bits/stdc++.h>
using namespace std;

string s;
vector <string> expr, op;
int res1, res2;

vector <string> split(string input, char delimiter){
    vector <string> result;
    stringstream ss(input);
    string tmp;

    while(getline(ss,tmp,delimiter)) result.push_back(tmp);
    return result;
}

int getRes(string sa, string op, string sb){
  int a = stoi(sa), b = stoi(sb);
  if(op == "+") return a + b;
  if(op == "-") return a - b;
  if(op == "*") return a * b;
  return a / b;
}

int main(){
  getline(cin, s);
  expr = split(s, ' ');

  res1 = getRes(expr[0], expr[1], expr[2]);
  res1 = getRes(to_string(res1), expr[3], expr[4]);

  res2 = getRes(expr[2], expr[3], expr[4]);
  res2 = getRes(expr[0], expr[1], to_string(res2));

  cout << min(res1, res2) << '\n' << max(res1, res2);
}